packages/libs/src/services/solana/SolanaWeb3Manager.ts

```diff
@@ -766,6 +780,55 @@ export class SolanaWeb3Manager {
     })
   }
 
+  async purchaseContentWithPaymentRouterUserBank({
+    id,
+    type,
+    blocknumber,
+    extraAmount = 0,
+    splits,
+    purchaserUserId,
+    skipSendAndReturnTransaction
```

Comment:
Could we not just split out the part the creates the transaction into its own funciton, and then make this function just do:
* Call the work function
* Send the transaction

It feels a little strange to have this function switch between returning a transaction and returning a result of submitting a transaction.

Reply:
yeah i agree. the problem is then we'd only have to add the extra signature in the sending from solana root account case. felt weird to me and was trying to think of something more elegant.. but maybe it's good enough.

Reply:
ah now i remember why it breaks down: we need the solana root account keypair in order to partialSign the tx, but we don't have/need the private key for the user bank case. We could pass the public and private key separately and have the private key be optional, but then we'd have to reconstruct the keypair before signing... gets ugly and not sure if better than what we have here.

Reply:
ok i tried it, not as bad as i thought